File: src/main/java/org/railwaystations/rsapi/app/RsapiApplicationRunner.java
package org.railwaystations.rsapi.app;

import org.railwaystations.rsapi.core.ports.in.GetStatisticUseCase;
import org.railwaystations.rsapi.core.ports.out.Monitor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

@Component
public class RsapiApplicationRunner implements CommandLineRunner {

    @Autowired
    private Monitor monitor;

    @Autowired
    private GetStatisticUseCase getStatisticUseCase;

    @Override
    public void run(String... args) {
        monitor.sendMessage(getStatisticUseCase.getCountryStatisticMessage());
    }

}

File: service/src/main/kotlin/org/railwaystations/rsapi/RsapiApplicationRunner.kt
package org.railwaystations.rsapi

import org.railwaystations.rsapi.core.ports.outbound.MonitorPort
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component

@Component
class RsapiApplicationRunner(
    private val monitorPort: MonitorPort,
) : CommandLineRunner {

    override fun run(vararg args: String) {
        monitorPort.sendMessage("RSAPI startup")
    }
}
